src/app/components/demo/demo.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit, Optional, Inject } from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+
+import { ConfigOptionsService } from 'src/app/modules/core/services/config-options-service/config-options.service';
+import { LocalStorageService } from 'src/app/modules/core/services/local-stogage-service/local-storage.service';
+import { ConstantsService, Constant } from 'src/app/modules/core/services/constants-service/constants.service';
+import { GeneratorService } from 'src/app/modules/core/services/generator-service/generator.service';
+
+@Component({
+  selector: 'app-demo',
+  templateUrl: './demo.component.html',
+  styles: []
+})
+export class DemoComponent implements OnInit {
+  constant = {};
+  storageValues = [];
+  configOptions = {};
+  sequence = '';
+  localStorageForm: FormGroup;
+  configOptionsForm: FormGroup;
+  noProviderForLocalStorage = '';
+  noProviderForOptionsService = '';
+
+  constructor(
+    @Optional() @Inject(ConstantsService) public constantsService: Constant,
+    @Optional() @Inject(GeneratorService) private generatorService: string,
+    @Optional() private configOptionsService: ConfigOptionsService,
+    @Optional() private localStorageService: LocalStorageService,
+    private formBuilder: FormBuilder,
+  ) { }
+
+  ngOnInit() {
+    if (!this.localStorageService) {
+      this.noProviderForLocalStorage = 'No Provider for LocalStorageService';
```

Comment:
this text: No Provider for LocalStorageService is better to extract into constants

Reply:
Done